core/swift3Action/spm-build/Whisk.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  Whisk.swift
+//  swift3Action
+//
+//  Created by Justin Berstler on 5/5/16.
+//
+//
+
+import Foundation
+import Dispatch
+import KituraNet
+import KituraSys
+
+class Whisk {
+    class func invoke(actionNamed action : String, withParameters params : [String:Any]) -> [String:Any] {
+        let parsedAction = parseQualifiedName(name: action)
+        let path = "/api/v1/namespaces/\(parsedAction.namespace)/actions/\(parsedAction.name)?blocking=true"
+
+        return postSyncronish(uriPath: path, params: params)
+    }
+
+    class func trigger(eventNamed event : String, withParameters params : [String:Any]) -> [String:Any] {
+        let parsedEvent = parseQualifiedName(name: event)
+        let path = "/api/v1/namespaces/\(parsedEvent.namespace)/triggers/\(parsedEvent.name)?blocking=true"
+
+        return postSyncronish(uriPath: path, params: params)
+    }
+
+    // handle the GCD dance to make the post async, but then obtain/return
+    // the result from this function sync
+    private class func postSyncronish(uriPath path: String, params : [String:Any]) -> [String:Any] {
+        var response : [String:Any]!
+
+        let invokeGroup = dispatch_group_create()
+        dispatch_group_enter(invokeGroup)
+
+        let queue = dispatch_queue_create("com.ibm.openwhisk.post.queue", DISPATCH_QUEUE_CONCURRENT)
+
+        dispatch_async(queue) {
+            post(uriPath: path, params: params) { result in
+                response = result
+                dispatch_group_leave(invokeGroup)
+            }
+        }
+
+        // On one hand, FOREVER seems like an awfully long time...
+        // But on the other hand, I think we can rely on the system to kill this
+        // if it exceeds a reasonable execution time.
+        dispatch_group_wait(invokeGroup, DISPATCH_TIME_FOREVER)
+        return response
+    }
+
+    /*
+     * Initialize with host, port and authKey determined from environment variables
+     * EDGE_HOST and AUTH_KEY, respectively
+     */
+    private class func initializeCommunication() -> (host : String, port : Int16, authKey : String) {
+        let env = NSProcessInfo.processInfo().environment
+
+        var edgeHost = "fakehost:80"
+        if let edgeHostEnv : String = env["EDGE_HOST"] {
```

Comment:
Seems safe to me to assert it will be in the env.? It will fail later otherwise anyway with a cryptic reference to `fakehost` otherwise, it seems.